targets/targets.json

```diff
@@ -8778,6 +8777,65 @@
         "mbed_ram_start"        : "0x20000000",
         "mbed_ram_size"         : "0x8000"
     },
+    "NU_PFM_M2351_NS": {
+        "inherits": ["NSPE_Target", "NU_PFM_M2351_CM"],
+        "core": "Cortex-M23-NS",
+        "tfm.level": 1,
+        "extra_labels_add": [
+            "M23_NS",
+            "PSA",
+            "TFM",
+            "NU_PREBUILD_SECURE"
+        ],
+        "macros_add": [
+            "MBED_TZ_DEFAULT_ACCESS=1",
+            "CMSIS_NVIC_VIRTUAL",
+            "MBEDTLS_PSA_CRYPTO_C"
+        ],
+        "components_add": ["FLASHIAP"],
+        "post_binary_hook": {"function": "M2351Code.merge_secure"},
+        "secure_image_filename": "tfm.hex",
+        "overrides": {
+            "secure-rom-start"      : "0x0",
+            "secure-rom-size"       : "0x3C000",
+            "secure-ram-start"      : "0x20000000",
+            "secure-ram-size"       : "0x10000",
+            "non-secure-rom-start"  : "0x1003C000",
+            "non-secure-rom-size"   : "0x44000",
+            "non-secure-ram-start"  : "0x30010000",
+            "non-secure-ram-size"   : "0x8000"
```

Comment:
Maybe drop the `non-secure-`-prefix?

Reply:
No. They are both defined in [PSA_Target](https://github.com/ARMmbed/mbed-os/blob/eeb033f8148a87659f3c084ec0b69e59ba9ac78e/targets/targets.json#L61).